--- a/vendor/xiaomi/sm8450-common/Android.bp
+++ b/vendor/xiaomi/sm8450-common/Android.bp
@@ -3961,61 +3961,61 @@
     },
     compile_multilib: "64",
     prefer: true,
     soc_specific: true,
 }
 
 cc_prebuilt_library_shared {
     name: "libgnsspps",
     owner: "xiaomi",
     strip: {
         none: true,
     },
     target: {
         android_arm64: {
             srcs: [
                 "proprietary/vendor/lib64/libgnsspps.so",
             ],
             shared_libs: [
                 "libutils",
                 "libcutils",
                 "libgps.utils",
                 "liblog",
                 "libc++",
                 "libc",
                 "libm",
                 "libdl",
             ],
         },
     },
     compile_multilib: "64",
-    prefer: true,
+    prefer: false,
     soc_specific: true,
 }
 
 cc_prebuilt_library_shared {
     name: "libgrpc++_unsecure_prebuilt",
     owner: "xiaomi",
     strip: {
         none: true,
     },
     target: {
         android_arm64: {
             srcs: [
                 "proprietary/vendor/lib64/libgrpc++_unsecure_prebuilt.so",
             ],
             shared_libs: [
                 "libz",
                 "liblog",
                 "libc++",
                 "libc",
                 "libm",
                 "libdl",
             ],
         },
     },
     compile_multilib: "64",
     prefer: true,
     soc_specific: true,
 }
 
 cc_prebuilt_library_shared {
