--- a/vendor/xiaomi/sm8450-common/Android.bp
+++ b/vendor/xiaomi/sm8450-common/Android.bp
@@ -4380,61 +4380,61 @@
     soc_specific: true,
 }
 
 cc_prebuilt_library_shared {
     name: "libloc_api_v02",
     owner: "xiaomi",
     strip: {
         none: true,
     },
     target: {
         android_arm64: {
             srcs: [
                 "proprietary/vendor/lib64/libloc_api_v02.so",
             ],
             shared_libs: [
                 "libutils",
                 "libcutils",
                 "libqmi_cci",
                 "libqmi_common_so",
                 "libloc_core",
                 "libgps.utils",
                 "liblog",
                 "libc++",
                 "libc",
                 "libm",
                 "libdl",
             ],
         },
     },
     compile_multilib: "64",
-    prefer: true,
+    prefer: false,
     soc_specific: true,
 }
 
 cc_prebuilt_library_shared {
     name: "libloc_api_wds",
     owner: "xiaomi",
     strip: {
         none: true,
     },
     target: {
         android_arm64: {
             srcs: [
                 "proprietary/vendor/lib64/libloc_api_wds.so",
             ],
             shared_libs: [
                 "libgps.utils",
                 "libidl",
                 "libcutils",
                 "liblog",
                 "libutils",
                 "liblocation_api",
                 "libqmi_cci",
                 "libqmi_common_so",
                 "libc++",
                 "libc",
                 "libm",
                 "libdl",
             ],
         },
     },
@@ -4593,61 +4593,61 @@
     },
     compile_multilib: "64",
     prefer: true,
     soc_specific: true,
 }
 
 cc_prebuilt_library_shared {
     name: "libloc_socket",
     owner: "xiaomi",
     strip: {
         none: true,
     },
     target: {
         android_arm64: {
             srcs: [
                 "proprietary/vendor/lib64/libloc_socket.so",
             ],
             shared_libs: [
                 "libutils",
                 "libcutils",
                 "liblog",
                 "libgps.utils",
                 "libc++",
                 "libc",
                 "libm",
                 "libdl",
             ],
         },
     },
     compile_multilib: "64",
-    prefer: true,
+    prefer: false,
     soc_specific: true,
 }
 
 cc_prebuilt_library_shared {
     name: "libloc_util",
     owner: "xiaomi",
     strip: {
         none: true,
     },
     target: {
         android_arm64: {
             srcs: [
                 "proprietary/vendor/lib64/libloc_util.so",
             ],
             shared_libs: [
                 "libutils",
                 "libcutils",
                 "liblog",
                 "libgps.utils",
                 "libc++",
                 "libc",
                 "libm",
                 "libdl",
             ],
         },
     },
     compile_multilib: "64",
     prefer: true,
     soc_specific: true,
 }
@@ -9254,61 +9254,61 @@
     soc_specific: true,
 }
 
 cc_prebuilt_library_shared {
     name: "libsynergy_loc_api",
     owner: "xiaomi",
     strip: {
         none: true,
     },
     target: {
         android_arm64: {
             srcs: [
                 "proprietary/vendor/lib64/libsynergy_loc_api.so",
             ],
             shared_libs: [
                 "libutils",
                 "libcutils",
                 "libqmi_cci",
                 "libqmi_common_so",
                 "libloc_core",
                 "libgps.utils",
                 "liblog",
                 "libc++",
                 "libc",
                 "libm",
                 "libdl",
             ],
         },
     },
     compile_multilib: "64",
-    prefer: true,
+    prefer: false,
     soc_specific: true,
 }
 
 cc_prebuilt_library_shared {
     name: "libsys_info_cache",
     owner: "xiaomi",
     strip: {
         none: true,
     },
     target: {
         android_arm64: {
             srcs: [
                 "proprietary/vendor/lib64/libsys_info_cache.so",
             ],
             shared_libs: [
                 "libutils",
                 "libcutils",
                 "libgps.utils",
                 "libloc_core",
                 "liblog",
                 "libc++",
                 "libc",
                 "libm",
                 "libdl",
             ],
         },
     },
     compile_multilib: "64",
     prefer: true,
     soc_specific: true,
